atomix/core/src/main/java/io/atomix/core/Atomix.java

```diff
@@ -109,7 +104,11 @@ protected Atomix(
         Executors.newScheduledThreadPool(
             Math.max(Math.min(Runtime.getRuntime().availableProcessors() * 2, 8), 4),
             Threads.namedThreads("atomix-primitive-%d", LOGGER));
-    partitions = buildPartitionService(config, getMembershipService(), getCommunicationService());
+
+    final PartitionGroupConfig<?> partitionGroupConfig = config.getPartitionGroup();
```

Comment:
:thought_balloon: : we might want to "merge" the `PartitionGroup` and the `RaftPartitionGroup` (and related classes) into a single interface. We won't support different types of partitions, but having an interface is still beneficial for testing. But maybe some of the RaftPartitionGroup specific properties might be lifted up? Just thinking out loud.